PowerEditor/src/WinControls/OpenSaveFileDialog/CustomFileDialog.cpp

```diff
@@ -130,22 +359,29 @@ class CustomFileDialog::Impl
 		return SUCCEEDED(hr);
 	}
 
-	bool setInitDir(const TCHAR* dir)
+	bool setFolder(const TCHAR* dir)
 	{
-		IShellItem* psi = nullptr;
-		HRESULT hr = SHCreateItemFromParsingName(dir,
-			0,
-			IID_IShellItem,
-			reinterpret_cast<void**>(&psi));
-		if (SUCCEEDED(hr))
-			hr = _dialog->SetFolder(psi);
-		return SUCCEEDED(hr);
+		return setDialogFolder(_dialog, dir);
 	}
 
 	bool show()
 	{
-		HRESULT hr = _dialog->Show(_hwndOwner);
-		return SUCCEEDED(hr);
+		bool okPressed = false;
+		HRESULT hr = FileDialogEventHandler::createInstance(IID_PPV_ARGS(&_events));
+		if (SUCCEEDED(hr))
+		{
+			DWORD dwCookie;
+			hr = _dialog->Advise(_events, &dwCookie);
```

Comment:
Should we check the preference that enables the Unix-style file path?

Reply:
>Should we check the preference that enables the Unix-style file path?

I don't think so, because if this all works out, that pref setting will be going away.  :-)